Add snakeyaml dependency to Yaml configuration feature

wetted opened the pull request:
closes #1514

wetted committed:
Add snakeyaml dependency to Yaml configuration feature

closes #1514

wetted committed:
Merge remote-tracking branch 'origin/3.9.x' into Issue1514-snakeyaml_for_3.9

sdelamo commented:
is this in draft or ready for review? Please, assign me as a reviewer when it is ready

sdelamo committed:
add dependencies only if visible

sdelamo committed:
Merge branch '3.9.x' into Issue1514-snakeyaml_for_3.9

sdelamo committed:
test: BuildTestVerifier API

This API allows to test dependencies in Gradle or Maven builds with the same api

sdelamo committed:
yaml test

sdelamo commented:
I merge https://github.com/micronaut-projects/micronaut-starter/pull/1590 into this PR.

sdelamo committed:
Merge branch '3.9.x' into Issue1514-snakeyaml_for_3.9

sdelamo committed:
Update YamlSpec.groovy

sdelamo commented:
ping @timyates

timyates commented on starter-core/src/test/groovy/io/micronaut/starter/feature/config/YamlSpec.groovy:
```diff
@@ -21,6 +22,12 @@ class YamlSpec extends BeanContextSpec implements CommandOutputFixture {
         yaml.order == FeaturePhase.HIGHEST.getOrder()
     }
 
+    @PendingFeature
+    void "is visible"() {
+        expect:
+        yaml.isVisible()
+    }
+
```
Is this to check it's not visible?

Or is it going to be visible one day?

If it's to check that `isVisible()` is false, and it's staying that way, we should remove `@PendingFeature` and just test that it is false

wetted replied:
This PR is for 3.9.0 and not visible. For 4.0.0 it will change to be visible, when the `snakeyaml` dependency will be added.

timyates replied:
👍
